src/GraphQL/Execution/IArrayPool.cs

```diff
@@ -0,0 +1,22 @@
+namespace GraphQL.Execution
+{
+    /// <summary>
+    /// Provides a resource pool of temporary arrays during query execution.
+    /// Can be used to return lists of data from field resolvers.
+    /// </summary>
+    /// <inheritdoc cref="System.Buffers.ArrayPool{T}"/>
+    public interface IArrayPool
```

Comment:
```suggestion
    public interface IExecutionArrayPool
```
I suggested `IExecutionArrayPool` because the naming helps reinforce to a user that it is for the lifetime of the execution, rather than a generic shared array pool.  It also matches the comments.  If you really don't like that, it's okay though.

Reply:
50/50 . It's already in `GraphQL.Execution` namespace so...  but on other hand I see this:
![изображение](https://user-images.githubusercontent.com/21261007/104846012-6e889080-58e9-11eb-856c-a3dc29ac4a1b.png)

😕

Reply:
If you’re 50/50, let’s change it to IExecutionArrayPool